### PR TITLE
Replace ioutil with os and io for the test/images

### LINKED ISSUE #100367: Fixes for Deprecation of io/ioutil (in Go 1.16)
Please see https://golang.org/doc/go1.16#ioutil

We have a lot of references to `ioutil` which we need to migrate to newer package(s) mentioned in the url above:
https://cs.k8s.io/?q=io%2Fioutil&i=nope&files=.*%5C.go&excludeFiles=&repos=kubernetes/kubernetes

We also need some verify scripts to ensure we don't add back this package

/area code-organization

### PR DESCRIPTION
**Pull request by ahrtr**
The ioutil package has already been deprecated in golang 1.16, please see [go1.16#ioutil](https://golang.org/doc/go1.16#ioutil). So we need to replace all the ioutil with os and io.

This PR is just to update the test/images. There will be a couple of separate PRs for another SIGs.

Previously there is a PR, which is huge. So I am splitting the huge PR into a couple of smaller PRs. This PR is just one of them.

Fix #100367

cc @spiffxp

### CODE REVIEW

File: build/dependencies.yaml [hunk 1]
```diff
@@ -215,7 +215,7 @@ dependencies:
 
   # metadata-concealment: bump this one first
   - name: "metadata-concealment"
-    version: "1.6"
+    version: "1.7"
```
Comment (dims) [inline build/dependencies.yaml:218]: unrelated change?

Comment (ahrtr) [reply build/dependencies.yaml]: Right, thanks for pointing out this. Just fixed.

**Comment by ahrtr** [comment]
Just rebased the PR.

File: test/images/metadata-concealment/check_metadata_concealment.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/*
```
Comment (oomichi) [inline test/images/metadata-concealment/check_metadata_concealment.go:1]: Why do we need this additional file?
I could not get the point from the pr message.

File: test/images/jessie-dnsutils/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-1.6
+1.7
```
Comment (mkumatag) [inline test/images/jessie-dnsutils/VERSION:1]: why did we bump the version here? I don't see any changes to this image

**Review by mkumatag — CHANGES_REQUESTED** [review]
can you bump the version for the agnhost image as well?

File: test/images/jessie-dnsutils/VERSION (hunk 3 above)
Comment (ahrtr) [reply test/images/jessie-dnsutils/VERSION]: Thanks for pointing it out. Just reverted.

**Comment by ahrtr** [comment]
Just resolved a comment & rebased this PR.

**Comment by mkumatag** [comment]
> Just resolved a comment & rebased this PR.

but I still don't see the version bump for the agnhost image

**Comment by ahrtr** [comment]
> > Just resolved a comment & rebased this PR.
> 
> but I still don't see the version bump for the agnhost image

Done. Thanks.

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve
